shell/testing/tester_main.cc

```diff
@@ -513,6 +524,84 @@ int RunTester(const flutter::Settings& settings,
   return completion_observer.GetExitCodeForLastError();
 }
 
+#ifdef _WIN32
+#define EXPORTED __declspec(dllexport)
+#else
+#define EXPORTED __attribute__((visibility("default")))
+#endif
+
+extern "C" {
+EXPORTED Dart_Handle LoadLibraryFromKernel(const char* path) {
+  std::shared_ptr<fml::FileMapping> mapping =
+      fml::FileMapping::CreateReadOnly(path);
+  if (!mapping) {
+    FML_LOG(ERROR) << "Failed to load file at \"" << path << "\".";
+    return Dart_Null();
```

Comment:
Where is it throwing an exception? We don't like throwing dart exceptions from C++ code in this repo because it causes hard to understand problems for users.

Reply:
Throwing an exception on the Dart side is fine. We want to avoid making any API calls that would fail to run C++ destructors (such as Dart_ThrowException or Dart_PropagateError). In general these API are avoided in the engine because it can quickly become hard to reason about what C++ code won't run under what conditions, and what state that will leave the process in. In particular, we want to avoid the process segfaulting even if it means more (recoverable) Dart exceptions.